File: backend/smart-parking-system/src/main/java/com/smartparking/service/AuthService.java
package com.smartparking.service;

import com.smartparking.dto.request.LoginRequest;
import com.smartparking.dto.request.RegisterRequest;
import com.smartparking.dto.response.LoginResponse;
import com.smartparking.dto.response.UserResponse;

public interface AuthService {

    UserResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

}

File: frontend/smart-parking-frontend/src/services/authService.js
import axiosInstance from "./axiosInstance";

export function register(registerRequest) {

    return axiosInstance({

        method: "POST",

        url: "/auth/register",

        data: registerRequest

    });

}

export function login(loginRequest) {

    return axiosInstance({

        method: "POST",

        url: "/auth/login",

        data: loginRequest

    });

}